spec/components/radio_spec.rb

```diff
@@ -148,6 +148,37 @@ def component_name
     assert_select "legend h1", "What is your favourite skittle?"
   end
 
+  it "renders radio-group with a legend" do
+    render_component(
+      name: "radio-group-legend",
+      heading: "What is your favourite smartie?",
+      heading_level: 0,
+      items: [
+        { label: "Red", value: "red" },
+        { label: "Blue", value: "blue" },
+      ],
+    )
+    assert_select ".govuk-radios"
+    assert_select "legend", "What is your favourite smartie?"
+    assert_select "legend span.govuk-fieldset__heading", true
+  end
+
+  it "renders radio-group with a custom legend size" do
```

Comment:
I'm not sure this test is necessary - testing the size is already done by the `it "renders radio-group with a custom page heading size"` and similar tests.

Reply:
I thought this was _slightly_ different as it checks the `legend > span` is being styled?

Reply:
Possibly. Could this be folded into one of the existing tests then?

Reply:
So with this one, there was no other test that had `heading_level` set to `0` as a result created a `span` I did think this was slightly different and couldn't quite roll this into another one.

Reply:
But the options passed to the component for this test exactly match the options in the `renders radio-group with a legend` test on line 151 (unless I've overlooked something). Could the assert_select lines from this test not be moved into that one?